### PR TITLE
test: add gateway command routing coverage

### LINKED ISSUE #89: test: gateway command routing coverage (/agents, /install, /start, /stop, /status, /logs, /upgrade, /diagnose)
## Summary
Gateway command routing for P0 commands lacks test coverage. Only `/diagnose-consent` and `parseInput` are tested.

## What to test
- `/agents` — lists agents with installed count
- `/install <agent_id>` — success + missing arg
- `/start <agent_id>` — success + already running + not installed
- `/stop <agent_id>` — success + already stopped
- `/status [agent_id]` — all agents vs single agent
- `/logs <agent_id> [tail]` — with/without tail param, download URL generation for large logs
- `/upgrade <agent_id>` — success + missing arg
- `/diagnose <agent_id>` — artifact + download URL
- Session required enforcement (non-`/pair` commands without session)
- Unknown command → ParseError

## Related
- `gateway/src/index.ts` (handleCommand, safeHandleCommand)

### PR DESCRIPTION
**Pull request by dev01lay2**
Closes #89

Adds tests for all gateway command routes: `/pair`, `/agents`, `/install`, `/start`, `/stop`, `/status`, `/logs`, `/upgrade`, `/diagnose`, plus session requirement enforcement and `parseInput` edge cases.

### CODE REVIEW

**Review by Akane-CN — APPROVED** [review]
Re-reviewed latest commit and reran checks.

- CI is green.
- No new blockers found.

Approving this revision.

File: gateway/src/command-routing.test.ts [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+describe("parseInput edge cases", () => {
+  test("unknown command throws", () => {
+    expect(() => parseInput("telegram 100 req-1 /unknown")).toThrow();
+  });
+
+  test("too few parts throws", () => {
+    expect(() => parseInput("telegram 100")).toThrow();
+  });
+});
```
[the remark this answers is not in the document]
Comment (dev01lay2) [reply gateway/src/command-routing.test.ts]: Good catch, fixed in 95985e3. Changed `toThrow()` to `toThrow(ParseError)` for both edge cases.
